Feat/expose built in ops partial update

aseure opened the pull request.

aseure committed:
feat(search): expose built-in operations for partial updates with PartialUpdateOperation<T>

aseure committed:
test(search): add serialization tests for PartialUpdateOperation<T>

aallam commented on algoliasearch-core/src/main/java/com/algolia/search/models/indexing/PartialUpdateOperation.java:
```diff
@@ -0,0 +1,53 @@
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class PartialUpdateOperation<T extends Object> {
```
No need to explicistly extend `Object`, we can replace `T extends Object` by `T`

aallam commented on algoliasearch-core/src/main/java/com/algolia/search/models/indexing/PartialUpdateOperation.java:
```diff
@@ -0,0 +1,53 @@
+    this.value = value;
+  }
+
+  public static PartialUpdateOperation<Integer> Increment(Integer value) {
```
Maybe we should avoid capital letters for method names?
(same applies to other methods below)

We can either use lowercase letters for the methods, or replace them with inner static classes:
```java
public static class Increment extends PartialUpdateOperation<Integer> {
    public Increment(Integer value) {
      super(PartialUpdateOperationType.INCREMENT, value);
    }
}
```

aseure replied:
Right, thanks a lot! I'm doing Go, Scala, C# and Java at the same time today so my casing game is a bit weird today ^^

I'll go with keeping the static functions if that's OK with you, I'm not a big fan of introducing inheritance for such a niche use-case.

aallam reviewed (approved):
Looks perfect to me 👍

aseure committed:
chore: fix fmt-maven-plugin configuration

Previously, the fmt-maven-plugin configuration was located under the
<build> configuration of our default profile. This lead to our
configured version of our formater to be completely ignored. Later on,
an error was discovered since a very recent version of our formater was
fetched instead, whose one of its transitive dependencies
(google-java-format) was upgraded (1.8) to a version (1.8) which drop
the Java 8 support.

This commit correctly sets up the fmt-maven-plugin configuration within
the <build><pluginManagement><plugins> tree instead. It also updates the
plugin from 2.8 to 2.9 and adds a comment to the pom.xml indicated *not
to* upgrade to 2.10, this version being the one whose transitive
dependencies drop Java 8 support.

aseure commented:
@Ant-hem @Aallam Just fixed the build with a third commit (see commit message for the details). Everything's still good to you?

aallam reviewed (approved):
Yup all good 👍
